controllers/routing/routing_reconcile_resources.go

```diff
@@ -117,6 +118,16 @@ func propagateHostsToWatchedCR(target *unstructured.Unstructured, data spi.Routi
 	return nil
 }
 
+func (r *PlatformRoutingController) ensureResourceHasFinalizer(ctx context.Context, target *unstructured.Unstructured) error {
+	if controllerutil.AddFinalizer(target, finalizerName) {
+		if err := cluster.Patch(ctx, r.Client, target); err != nil {
```

Comment:
If we do the `Patch` instead of `Update` we need to make sure we have right roles on the consumer side - odh-operator.